x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/result_settings_logic.test.ts

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter } from '../../../../__mocks__';
+
+import { Schema, SchemaConflicts, SchemaTypes } from '../../../../shared/types';
+import { OpenModal, ServerFieldResultSettingObject } from '../types';
+
+import { ResultSettingsLogic } from '.';
+
+describe('ResultSettingsLogic', () => {
+  const { mount } = new LogicMounter(ResultSettingsLogic);
+
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    saving: false,
+    openModal: OpenModal.None,
+    nonTextResultFields: {},
+    resultFields: {},
+    serverResultFields: {},
+    textResultFields: {},
+    lastSavedResultFields: {},
+    schema: {},
+    schemaConflicts: {},
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const itShouldCloseOpenModals = (callback: () => void) => {
+    mount({
+      openModal: OpenModal.ConfirmSaveModal,
+    });
+
+    callback();
+
+    expect(ResultSettingsLogic.values).toEqual({
+      ...DEFAULT_VALUES,
+      openModal: OpenModal.None,
+    });
+  };
+
+  it('has expected default values', () => {
+    mount();
+    expect(ResultSettingsLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('initializeResultFields', () => {
+      const serverResultFields: ServerFieldResultSettingObject = {
+        foo: { raw: { size: 5 } },
+        bar: { raw: { size: 5 } },
+      };
+      const schema: Schema = {
+        foo: 'text' as SchemaTypes,
+        bar: 'number' as SchemaTypes,
+        baz: 'text' as SchemaTypes,
+      };
+      const schemaConflicts: SchemaConflicts = {
+        foo: {
+          text: ['foo'],
+          number: ['foo'],
+          geolocation: [],
+          date: [],
+        },
+      };
+
+      it('will initialize all result field state within the UI, based on provided server data', () => {
+        mount({
+          dataLoading: true,
+          saving: true,
+          openModal: OpenModal.ConfirmSaveModal,
+        });
+
+        ResultSettingsLogic.actions.initializeResultFields(
+          serverResultFields,
+          schema,
+          schemaConflicts
+        );
+
+        expect(ResultSettingsLogic.values).toEqual({
```

Comment:
This probably could have been like 10 different individual tests. For simplicity, I kept one, single test, that documents that covers the happy path for this action.